target/linux/generic/files/drivers/mtd/mtdsplit/mtdsplit_uimage.c

```diff
@@ -262,6 +317,33 @@ static struct mtd_part_parser uimage_generic_parser = {
 	.type = MTD_PARSER_TYPE_FIRMWARE,
 };
 
+static int
+mtdsplit_uimage_parse_asus(struct mtd_info *master,
+			      const struct mtd_partition **pparts,
+			      struct mtd_part_parser_data *data)
+{
+	return __mtdsplit_parse_uimage(master, pparts, data,
+				      HEADER_TYPE_ASUS,
+				      uimage_verify_default);
+}
+
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(4, 9, 0)
```

Comment:
OpenWrt currently supports 4.14/4.19/5.4, so this "if" is not necessary.

Reply:
It is for consistency: all or nothing and I did not dare to remove all kernel version checks from that file without having an ability to test all use cases involved

Reply:
Keep this open, maybe I'll do it separately after you patch has been merged, so it may serve as a reminder to me.

Reply:
I've dropped the kernel version switches here:
https://github.com/openwrt/openwrt/commit/8ff813e5f45169d0b97f406643197a62ff031b8d

So you can drop them now as well. :-)

Reply:
Done